### DISCUSSION TITLE
Is relationManager compatible with the BelongsToMany's orderByPivot method?

### DISCUSSION THREAD

**Discussion by gilr** [filamentphp/filament#8022] [Help]
Hi,
Without ->orderByPivot('order','asc') in a belongsToMany model relation, the attachment action works well.

When I add the orderByPivot, when trying to select a record, I get a SQL error as 'Distinct' clause is incompatible with the absence of the 'order by' field in the select list in SELECT.

`_SQLSTATE[HY000]: General error: 3065 Expression #1 of ORDER BY clause is not in SELECT list, references column project.collection_publication.order' which is not in SELECT list; this is incompatible with DISTINCT_`

`_SQL: select distinct `publications`.* from `publications` left join `collection_publication` on `publications`.`id` = `collection_publication`.`publication_id` where (`publications`.`name` like %<pattern>%) and not exists (select * from `collections` inner join `collection_publication` on `collections`.`id` = `collection_publication`.`collection_id` where `publications`.`id` = `collection_publication`.`publication_id` and `collections`.`id` = 89 and `collections`.`deleted_at` is null) and `publications`.`deleted_at` is null order by `collection_publication`.`order` asc`_

I tried the SQL request, and It works indeed with the addition of "`, `collection_publication`.order" (or .*) after the "Select distinct Select distinct `publications`.* from `publications"

Is there a solution, or do I have to remove my OrderByPivot clauses?

**Comment by johnnywo** [comment]
... I have exactly the same problem

**Reply by leandrocfe** [reply]
what about `->defaultSort('pivot_table_name.column', 'desc')` | 'asc' ?

**Reply by johnnywo** [reply]
.. yes thanks - this is the solution for the moment - but as I like to keep this default sorting also for frontend pages it would be appreciated to put it directly on the model relationship via ->orderByPivot() method

**Comment by AlexisSerneels** [comment]
Same here. May wanna send a bug report.

**Comment by edwwaarrdd** [comment]
Had the same problem.
I opened a PR which I think will fix this problem. You can follow along here: https://github.com/filamentphp/filament/pull/12308
